trino/trino.go

```diff
@@ -307,7 +308,9 @@ func newConn(dsn string) (*Conn, error) {
 		}
 	}
 
-	var httpClient = http.DefaultClient
+	var httpClient = &http.Client{
+		Transport: otelhttp.NewTransport(http.DefaultTransport),
```

Comment:
If users can pass a custom http client, we don't really need to use the OTEL transport by default.